src/template-tags/tickets.php

```diff
@@ -192,20 +193,24 @@ function tribe_events_count_available_tickets( $event = null ) {
 	function tribe_tickets_buy_button( $echo = true ) {
 		$event_id = get_the_ID();
 
-		// check if there are any tickets on sale
-		if ( ! tribe_events_has_tickets_on_sale( $event_id ) ) {
-			return null;
+		if ( empty( $event_id ) ) {
+			return '';
+		}
+
+		// check if there are any tickets available
+		if ( ! tribe_tickets_is_current_time_in_date_window( $event_id ) ) {
+			return '';
 		}
 
 		// get an array for ticket and rsvp counts
 		$types = Tribe__Tickets__Tickets::get_ticket_counts( $event_id );
 
 		// if no rsvp or tickets return
 		if ( ! $types ) {
-			return null;
+			return '';
 		}
 
-		$html = [];
+		$html  = [];
```

Comment:
:no_entry_sign: Short array syntax is not allowed

<sub>hash: 9deff1cc9527e8b82b96c96f8c5ad379</sub>